woocommerce/woocommerce-gutenberg-products-block
# Move `FormattedMonetaryAmount` to the components package

## What

This PR is based on https://github.com/woocommerce/woocommerce-blocks/pull/11214 (update/checkbox-component)

Moves the `FormattedMonetaryAmount` to the components package and updates imports used across the project to reflect the new location:

- `ProductSaleBadge`
- `renderPackageRateOption` in `ShippingRatesControlPackage`
- `TotalsFooterItem`
- `PriceSlider`
- `ProductPrice`
- `checkout-pickup-options-block`
- `checkout-shipping-method-block`
- `checkout-shipping-methods-block`
- `TotalsItem` (This is not imported from `@woocommerce/blocks-components` but it does use a relative import because of circular dependecies)

## Why

As more developers integrate their extensions with WooCommerce Blocks, the case for reusing components grows. Rather than recreating components developers can access `wc.blocksComponents` and get the ones we need.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Set a product to be on sale in your store.
2. Install [Multiple Packages for WooCommerce](https://wordpress.org/plugins/multiple-packages-for-woocommerce/) and go to WooCommerce > Settings > Multiple Packages. Set the "Group by" option to Product (individual).
3. Set up local pickup (WooCommerce -> Settings -> Shipping -> Local Pickup) and add a couple of locations. Add a price to this.
4. Set up a couple of shipping options with a price.
5. Add the `Filter by Price` and `Products (Beta)` blocks to a page on your site.
6. Go to the page and ensure the Filter by Price block is displaying the correct range of product prices.
7. Check the product that you set to be on sale, it should be showing the correct sale price, and all other products should be showing the correct price too.
8. Add 2 products to your cart.
9. Go to the Cart block. Ensure all products listed in the order summary sidebar show the correct prices.
10. Ensure the shipping options shown in the Cart block show the correct prices.
11. Ensure all totals in the Cart block are show the correct price.
12. Go to the Checkout block.
13. Ensure the Local Pickup/Shipping selector shows the correct "From" price in the buttons
<img width="764" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5656702/39906db8-9833-4a83-8275-11875a479ca1">

14. Select Shipping and scroll down to the shipping section. Ensure all shipping options show the correct price.
15. Change to Local Pickup and ensure the correct price is shown alongside the pickup location's name.
16. Ensure the totals are correct in the checkout sidebar. 
17. Repeat steps 5, 6, 7 but with the `All Products block`.


## Screenshots or screencast



N/A

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Make the `FormattedMonetaryAmount` component available in the `@woocommerce/blocks-components` package.
